aws-iot-dimension/src/main/java/com/amazonaws/iot/dimension/UpdateHandler.java

```diff
@@ -0,0 +1,149 @@
+    }
+
+    @Override
+    public ProgressEvent<ResourceModel, CallbackContext> handleRequest(
+            AmazonWebServicesClientProxy proxy,
+            ResourceHandlerRequest<ResourceModel> request,
+            CallbackContext callbackContext,
+            Logger logger) {
+
+        ResourceModel desiredModel = request.getDesiredResourceState();
+        String previousArn = request.getPreviousResourceState().getArn();
+        if (!Objects.equals(desiredModel.getArn(), previousArn)) {
+            logger.log(String.format("Arn cannot be updated, caller tried changing %s to %s.",
+                    previousArn, desiredModel.getArn()));
+            return ProgressEvent.failed(desiredModel, callbackContext, HandlerErrorCode.InvalidRequest,
+                    "Arn cannot be updated.");
+        }
+
+        String resourceArn = updateDimension(proxy, desiredModel, logger);
+
+        // For an exiting resource, we have to update via TagResource API, UpdateDimension API doesn't take tags.
+        updateTags(proxy, request, resourceArn, logger);
+
+        logger.log(String.format("Successfully updated %s.", resourceArn));
+        return ProgressEvent.defaultSuccessHandler(request.getDesiredResourceState());
+    }
+
+    /**
+     * @return The dimension's ARN.
+     */
+    private String updateDimension(AmazonWebServicesClientProxy proxy,
+                                   ResourceModel model,
+                                   Logger logger) {
+
+        UpdateDimensionRequest updateDimensionRequest = UpdateDimensionRequest.builder()
+                .name(model.getName())
+                .stringValues(model.getStringValues())
+                .build();
+
+        UpdateDimensionResponse updateDimensionResponse;
+        try {
+            updateDimensionResponse =
+                    proxy.injectCredentialsAndInvokeV2(updateDimensionRequest, iotClient::updateDimension);
+        } catch (IotException e) {
+            throw Translator.translateIotExceptionToCfn(e);
+        }
+        String arn = updateDimensionResponse.arn();
+        logger.log(String.format("Called UpdateDimension for %s.", arn));
+        return arn;
+    }
+
+    @VisibleForTesting
+    void updateTags(AmazonWebServicesClientProxy proxy,
+                    ResourceHandlerRequest<ResourceModel> request,
+                    String resourceArn,
+                    Logger logger) {
+
+        // Note: we're intentionally getting currentTags by calling ListTags rather than getting
+        // the previous state from CFN. This is in order to overwrite out-of-band changes.
+        // For example, if we used request.getPreviousResourceTags instead of ListTags, if a user added a new tag
+        // via TagResource and didn't add it to the template, we wouldn't know about it and wouldn't untag it.
+        // Yet we should, otherwise the resource wouldn't equate the template.
+        Set<Tag> currentTags = listTags(proxy, resourceArn, logger);
```

Comment:
We need to verify if the aws prefixed tags provided by CFN will be sent as result of listTags.

Reply:
Agreed, noted this on our integration test plan.